teps/0157-enhance-results-cli.md

```diff
@@ -0,0 +1,156 @@
+
+### Goals
+
+- Provide a user-friendly CLI for interacting with Tekton Results.
+- Improve accessibility and usability over the existing CLI.
+- Enable users to retrieve and analyze PipelineRun and TaskRun results efficiently.
+- Ensure authentication mechanisms are secure and easy to configure.
+- Allow the CLI to function both as a standalone tool and as a plugin to tkn CLI.
+
+### Non-Goals
+
+This proposal is not intended to modify the archived PipelineRuns and TaskRuns
+
+### Use Cases
+
+- As a `Developer`, I want to `list` my past PipelineRuns and TaskRuns without writing custom API queries.
+- As a `Tekton admin`, I want to `describe` specific runs to debug failures.
+- As a `DevOps engineer`, I want to `fetch logs` of past runs.
+- As a `Tekton user`, I want to use the CLI as a plugin within `tkn` to keep all pipeline-related commands under a single tool.
+
+## Requirements
+
+### Functional Requirements
+
+- Retrieve a `list` of PipelineRuns and TaskRuns from Tekton Results.
+- Describe a `specific` PipelineRun or TaskRun with details.
+- Fetch `logs` of a PipelineRun or TaskRun.
+- Authenticate with the `Tekton Results API` using token-based authentication.
+- Support filtering results based on `namespace` (`-n` flag).
+
+### Non-Functional Requirements
+
+- Consistent command syntax, following `tkn` CLI conventions.
+- Formatted output, similar to the tkn CLI.
+- Provide clear and actionable `error messages`.
+
+## Proposal
+
+This proposal introduces a new CLI tool, `tkn-results`, designed to simplify interaction with Tekton Results. The CLI will provide commands to list, describe, and retrieve logs for `PipelineRuns` and `TaskRuns` stored in the Results API. Additionally, it will include authentication support to ensure secure access. The goal is to improve usability and efficiency over the existing CLI while maintaining a familiar command structure for Tekton users. The CLI will also be designed to work as a `plugin` to `tkn`, allowing users to execute commands seamlessly from within their existing Tekton CLI environment.
+
+## Design Details
+
+### Commands
+
+#### Configuration
+
+```sh
+tkn-results config set        # Set authentication and configuration parameters
+```
+
+#### Fetching PipelineRuns and TaskRuns
+
+```sh
+tkn-results pr ls -n <namespace>  # Get list of PipelineRuns in a namespace  
+tkn-results tr ls -n <namespace>  # Get list of TaskRuns in a namespace  
+```
+
+#### Describing Runs
+
+```sh
+tkn-results pr describe <pr-name> -n <namespace>  # Get details of a specific PipelineRun  
+tkn-results tr describe <tr-name> -n <namespace>  # Get details of a specific TaskRun  
+```
+
+#### Fetching Logs
+
+```sh
+tkn-results pr logs <pr-name> -n <namespace>  # Fetch logs of a specific PipelineRun  
+tkn-results tr logs <tr-name> -n <namespace>  # Fetch logs of a specific TaskRun  
+```
```

Comment:
I imagine these would be mostly equivalent to `tkn pr ls`, `tkn pr describe`, `tkn pr logs`?

It would be great if they supported (most of) the same CLI flags, so that I could just take a regular `tkn ...` command, change it to `tkn results ...` and have it work. But I wouldn't necessarily expect that to be the case in the first iteration.

Reply:
Eventually, if the `tkn-results` commands become fully compatible with the equivalent `tkn` commands, maybe the the regular commands could become results-aware?

So I wouldn't need to try both `tkn pr logs` and `tkn results pr logs` -- `tkn pr logs` would "just work"?

Reply:
> I imagine these would be mostly equivalent to `tkn pr ls`, `tkn pr describe`, `tkn pr logs`?

Thanks for taking a look @chmeliik 
Yes, that's what we have planned right now. I should eventually add those flags based on the requirements.

> Eventually, if the `tkn-results` commands become fully compatible with the equivalent `tkn` commands, maybe the the regular commands could become results-aware?
> 
> So I wouldn't need to try both `tkn pr logs` and `tkn results pr logs` -- `tkn pr logs` would "just work"?

I agree with your point, I think that will be a part of long-term goals for the `tkn`

Reply:
Would `tkn results` _only_ look at the stored results or would it also be aware of the data in etcd? If the subcommand can work for either, then feature parity definitely makes sense.

Reply:
I think as of now, `tkn-results` will look for data in the stored results but I think moving forward long term goal would be to have the functionality in `tkn` where it can look for data stored in `etcd` as well as the results.
cc @vdemeester